CRR/ReplicationStatusUpdater.js

```diff
@@ -284,15 +292,17 @@ class ReplicationStatusUpdater {
      * @returns {void}
      */
     _markPending(bucket, versions, cb) {
-        const options = { Bucket: bucket };
         waterfall([
-            next => this.s3.getBucketReplication(options, (err, res) => {
-                if (err) {
-                    this.log.error('error getting bucket replication', { error: err });
-                    return next(err);
-                }
-                return next(null, res.ReplicationConfiguration);
-            }),
+            next => {
+                this.s3.send(new GetBucketReplicationCommand(
+                    { Bucket: bucket },
+                ))
+                    .then(res => next(null, res.ReplicationConfiguration))
+                    .catch(err => {
+                        this.log.error('error getting bucket replication', { error: err });
+                        return next(err);
+                    });
+            },
```

Comment:
```suggestion
           async() => {
                const res = await this.s3.send(new GetBucketReplicationCommand(
                    { Bucket: bucket },
                ));
                
                return res.ReplicationConfiguration
            },
```

You can keep the `this.log.error('error getting bucket replication', { error: err });` if you want with a try catch

Reply:
good catch I'm gonna use this pattern instead